pkg/windows/windows.go

```diff
@@ -166,11 +165,25 @@ var (
 	}
 )
 
-// getFilesToTransfer returns the properly populated filesToTransfer map. Note this does not include the WICD binary.
-func getFilesToTransfer() (map[*payload.FileInfo]string, error) {
-	if filesToTransfer != nil {
-		return filesToTransfer, nil
+// createPayload returns the map of files to transfer with generated file info
+func createPayload(platform *config.PlatformType) (map[*payload.FileInfo]string, error) {
+	srcDestPairs, err := getFilesToTransfer(platform)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get list of files to transfer: %w", err)
 	}
+	files := make(map[*payload.FileInfo]string)
+	for src, dest := range srcDestPairs {
+		f, err := payload.NewFileInfo(src)
+		if err != nil {
+			return nil, fmt.Errorf("could not create FileInfo object for file %s: %w", src, err)
+		}
+		files[f] = dest
+	}
+	return files, nil
+}
+
+// getFilesToTransfer returns the properly populated filesToTransfer map. Note this does not include the WICD binary.
+func getFilesToTransfer(platform *config.PlatformType) (map[string]string, error) {
 	srcDestPairs := map[string]string{
 		payload.GcpGetValidHostnameScriptPath:  remoteDir,
```

Comment:
This PR could be easily reframed as "Copy only specific setup files given target platform" to avoid copying `payload.GcpGetValidHostnameScriptPath` to all platforms as it is specific to GCP.

ATM is out of the scope of the ticket, consider including it or a follow-up PR works too.